### PR TITLE
Support multiple ReturnValueTypes for func props

### PR DESCRIPTION
**Pull request by mpreyskurantov**

### CODE REVIEW

**Review by OlegKipchatov — APPROVED** [review]

File: packages/devextreme-react-generator/src/generator.ts [hunk 1]
```diff
@@ -139,9 +139,14 @@ export function getComplexOptionType(
...
-        functionDescriptor.returnValueType && (formatTypeDescriptor(functionDescriptor.returnValueType) || (functionDescriptor.returnValueType.type === 'void' && 'void'))
-      ) || BaseTypes.Any;
+      let returnType = '';
+      if (functionDescriptor.returnValueTypes) {
+        returnType = functionDescriptor.returnValueTypes.length > 1
+          ? functionDescriptor.returnValueTypes.map((t) => formatTypeDescriptor(t)).join(' | ')
+          : (formatTypeDescriptor(functionDescriptor.returnValueTypes[0]) || (functionDescriptor.returnValueTypes[0].type === 'void' && 'void')) || BaseTypes.Any;
+      } else {
+        returnType = BaseTypes.Any;
+      }
```
Comment (IlyaKhD) [inline packages/devextreme-react-generator/src/generator.ts:149]: **Suggestion**
Can this be simplified?


```suggestion
        returnType = functionDescriptor.returnValueTypes
          ? functionDescriptor.returnValueTypes.map((t) => formatTypeDescriptor(t)).join(' | ')
          :  BaseTypes.Any;
```

File: packages/devextreme-react-generator/src/generator.test.ts [hunk 2]
```diff
@@ -1258,4 +1258,114 @@ describe('mapWidget', () => {
...
+    ];
+
+    const expected = '(() => string)';
+
+    it('should return single type', () => {
+      expect(getComplexOptionType(types)).toEqual(expected);
+    });
+  });
+
+  describe('getComplexOptionType for func with void return type', () => {
```
[the remark this answers is not in the document]
Comment (mpreyskurantov) [reply packages/devextreme-react-generator/src/generator.test.ts:1309]: Yes, a lot of func props have single "void" ret type and require this code.

File: packages/devextreme-react-generator/src/generator.test.ts [hunk 3]
```diff
@@ -1258,4 +1258,114 @@ describe('mapWidget', () => {
...
+        type: 'Function',
+        acceptableValues: [],
+        isCustomType: false,
+        importPath: '',
+        isImportedType: false,
+        params: [],
+      },
+    ];
+
+    const expected = '(() => any)';
```
[the remark this answers is not in the document]
Comment (mpreyskurantov) [reply packages/devextreme-react-generator/src/generator.test.ts:1365]: In short, yes.

File: packages/devextreme-react-generator/src/generator.ts (hunk 1 above)
Comment (mpreyskurantov) [reply packages/devextreme-react-generator/src/generator.ts:149]: No. See prev.

**Review by IlyaKhD — DISMISSED** [review]

**Review by IlyaKhD — DISMISSED** [review]

**Review by IlyaKhD — DISMISSED** [review]

**Pull request merged**